VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -83,16 +73,167 @@ public void setData(IComponent component){
+                    EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.STARTED));
+                }
+            }
+        }
+
+        dbStore.getWatchedStateForVideoId(videoData.videoId,
+            new DataCallback<DownloadEntry.WatchedState>(true) {
+                @Override
+                public void onResult(DownloadEntry.WatchedState result) {
+                    DownloadEntry.WatchedState ws = result;
+                    if(ws == null || ws == DownloadEntry.WatchedState.UNWATCHED) {
+                        viewHolder.rowType.setTextColor(context.getResources().getColor(R.color.cyan_3));
+                    } else if(ws == DownloadEntry.WatchedState.PARTIALLY_WATCHED) {
+                        viewHolder.rowType.setTextColor(context.getResources().getColor(R.color.cyan_2));
+                    } else {
+                        viewHolder.rowType.setTextColor(context.getResources().getColor(R.color.grey_3));
+                    }
+                }
+                @Override
+                public void onFail(Exception ex) {
+                    logger.error(ex);
+                }
+            });
+
+        dbStore.getDownloadedStateForVideoId(videoData.videoId,
+            new DataCallback<DownloadEntry.DownloadedState>(true) {
+                @Override
+                public void onResult(DownloadEntry.DownloadedState result) {
+                    DownloadEntry.DownloadedState ds = result;
+                    if(ds == null || ds == DownloadEntry.DownloadedState.ONLINE) {
+                        // not yet downloaded
+                        viewHolder.bulkDownloadVideos.setVisibility(View.VISIBLE);
+                        viewHolder.bulkDownloadVideos.setOnClickListener(new View.OnClickListener() {
+                            @Override
+                            public void onClick(View v) {
+                                EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.STARTED));
+                                logger.debug("Download Button Clicked");
+                                //notifyDataSetChanged();
+                                download(videoData);
+                            }
+                        });
+                    } else if(ds == DownloadEntry.DownloadedState.DOWNLOADING) {
+                        // may be download in progress
+                        EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.STARTED));
+                        viewHolder.bulkDownloadVideos.setVisibility(View.GONE);
+                        storage.getDownloadProgressByDmid(videoData.dmId, new DataCallback<Integer>(true) {
+                            @Override
+                            public void onResult(Integer result) {
+                                if(result>=0 && result < 100){
+                                    EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.STARTED));
```

Comment:
right now, we just have NOT_STARTED, STARTED, PAUSED, COMPLETED. 
the logic here is - if video is marked as downloading-started in the local database, we just broadcast the message indicating that "download started"